### ISSUE TITLE
Analyse feedback from WCAG 2.2 prototype crit

### ISSUE THREAD

**Issue by CharlotteDowns** [alphagov/govuk-design-system#3303]
## What

This analyses and prioritises the work from [Organise and run a team crit of updated prototype #3245](https://github.com/alphagov/govuk-design-system/issues/3245).

## Why

So that teams using the design system can be compliant with WCAG 2.2.

## Who needs to work on this
Charlotte, Calvin

## Who needs to review this
Steve

## Done when
* [x] Read all feedback
* [x] Thank participants
* [x] Determine which feedback we've received that we want to pull into the deliverable
* [x] Add brief for required changes as GitHub comments in #3304  
* [x] Share with team

**Comment by CharlotteDowns** [comment]
# List of feedback with items selected that we will prioritise a solution for in the deliverable.

## Feedback on: Overview of WCAG 2.2-related changes in Summary list

- [ ] Is Accessibility considerations for Summary list' too generic a heading? Does it get across that it's 1. new 2. to do with WCAG 2.2
- [ ] The text highlighting worked really well for me – made it really clear what the link was referring to. (On Chrome on a Macbook Pro 🤷🏻‍♂️)
- [ ] The text highlight appears to keep the old text highlight if you click on a second link. Might cause confusion?

## Feedback on: Guidance about how to make Summary list comply with WCAG 2.2

- [ ] From beeps re: text fragments. Noticed that these two links use text fragments to link to other parts of the current page. Text fragments are not currently supported in Firefox (and will never be supported in IE 11, which we still technically support). In these browsers, the user will be jumped back to the top of the page instead. These should probably be swapped out for hardcoded IDs/anchors to ensure support for all of our browsers. This wouldn't do the text highlighting effect that fragments do, but this can be mimicked using :target CSS styles. Using IDs would also mean that the fragment wouldn't need to be kept in sync with any changes to the actual text, making them a little easier to maintain. An extra assumed benefit, is that it makes sharing a URL to that specific bit of guidance way easier. It would just be the summary list URL with a "#section-name" at the end instead of the massive text fragment URL format.
- [ ] Thoughts on the tag within the paragraph. I'm kinda smuggling this feedback from Nikin who spotted this when he popped by to say hello at the office the other day: The tag being in the paragraph feels a bit tight. As a suggestion, we could add a line break after the tag to make the text a little bit easier to parse?
- [ ] Summary list missing sentence in highlighted text? I don't know if you have the most recent content changes, so this might not be relevant atm. The following sentence is also part of the WCAG 2.2 for summary list: "If a user decides to go back to a previous answer through a card or row action, make sure information they have already entered is pre-populated." It's for 'redundant entry' with the security warning etc as the second section. That would mean that both summary list 2.2 changes are back to back - I don't know if this is ok or not. It might make them easier to find or cause confusion.
- [ ] Have we considered how this will look when we update the tag component? We're going to update it as part of the task list work, it'll be in sentence case, use subtler colours and lighter, larger text. 

## Feedback on: Overview components and patterns affected by WCAG 2.2

- [ ] Thoughts on the lists of things. 2 things come to mind: 1. What do we think about putting the lists in tables? I wonder if it'd be easier to ingest the info. Right now it feels a bit bunged up together to me. Mia made a similar suggestion about the typography guidance prototype and I think it's worked really well. 2. I'm wondering if we should set the individual criteria to be linkable to via Beeps' anchor links idea and and then format them to reflect that. So a way this could look is that you link to the summary list, and then the criteria (it's only one in the prototype but it should be 2?) could be 2 links in an indented sub-list. You could also fold this into the tables idea if we think that's good.
- [ ] More than one 2.2 criteria for some components. How do we deal with components that have more than one success criteria listed? For example, summary list has target size and redundant entry. Would a simple comma list be ok?
- [ ] It does feel like those lists of affected components/patterns could be structured to make them easier to read. Not sure the best way of doing it without knowing more about how users will approach the task, but some ideas: 1. Create a H4 for each of the success criteria, with relevant things linked below (this would mean repeating links for components that meet multiple criteria, not sure how critical a problem that is) 2. Some kind of table with checks in against relevant criteria, like a product comparison (feels weird) 3. A way to show the extent and severity of the issues against each (we've internally ranked this right? depends how responsible we think it is to mark some things as more serious than others)

## Feedback on: Guidance about work needed on a service to comply with WCAG 2.2

- [ ] We can put the exact month/year of publication and mandatory adherence now 🙌🏻
- [ ] If people think we own WCAG 2.2... I think we've been very clear that we're not the ones who own WCAG 2.2, but if there is any feedback to the contrary, we might need to be even more specific and say that it's NOT us as well as who it actually is.
- [ ] Should we ask teams to feed their experiences back to us? e.g. as a step in 'what you need to do' so we find out about any problems people have, and can update our guidance accordingly.

# What we're going to do

## Definitely

- Use our own way of highlighting content using classes and `:target`, applying a blue inset text to the tagged content from a link. Removing the use of text fragments.
- Revisit content of heading in the WCAG 2.2 summary on each component/pattern page affected.

## Maybe

- Run a quick experiment with the proposed default tag.
- Putting the list of components/patterns affected in a table.
- Add that we do not own WCAG 2.2.

**Comment by calvin-lau-sig7** [comment]
The question about organising the '[Components/Patterns affected in the Design System](https://deploy-preview-3090--govuk-design-system-preview.netlify.app/accessibility/wcag-2.2/#components-affected-in-the-design-system)' is interesting.

Could add an item:
- Investigate options how to organise list (by criteria? by component or patterns as-is? add subcategories?)

**Comment by stevenjmesser** [comment]
What's the use case or scenario? How does it fit into a user's journey? I'm guessing people would want to see which elements from their services are affected without having to look through every guidance page. 

If it's not too complicated to add and wouldn't upset the page design too much, it could be simple to add, monitor via analytics and iterate if needed.

**Comment by CharlotteDowns** [comment]
## Responses

@calvin-lau-sig7 are you proposing we add that to 'Definitely' or 'Maybe'. I'm in favour of us working on it.

## Experiment

A quick mock up of what the new tag design would look like, I think when the update happens we should not only revisit the tag but the colour of the inset text border too.

<img width="823" alt="New tag component in sentence case, with lighter font weight and new text and background colours" src="https://github.com/alphagov/govuk-design-system/assets/56260216/b8447084-a7f9-4580-908c-b830c135121f">

**Comment by calvin-lau-sig7** [comment]
> What's the use case or scenario? How does it fit into a user's journey? I'm guessing people would want to see which elements from their services are affected without having to look through every guidance page.

Yes, exactly. My assumption is that a service team would want an index to assess WCAG 2.2 work needed in on their service.

This is putting together 2, maybe 3 versions of the list and either deciding as a team or possibly A/B testing with users somehow.
